amundsen_application/static/js/components/common/EditableSection/index.tsx

```diff
@@ -118,15 +118,15 @@ export class EditableSection extends React.Component<
 
     return (
       <section className="editable-section">
-        <label className="editable-section-label">
+        <div className="editable-section-label">
```

Comment:
This element is intended to a be `label`. Previous discussion: https://github.com/lyft/amundsenfrontendlibrary/pull/480#discussion_r441140060

Can you explain this fix and why you think this works? When making fixes we are interested in debugging the source of the problem, and fixing the issue at the source -- even if other hacks or changes are successful in making the problem go away, we're always concerned about making the best fix possible. 

I would revisit our exploration of event propagation, and the fact that we say that clicking on a child in `EditableSection` was triggering `toggleEdit` when it wasn't supposed to.

Reply:
While I was debugging this and monitoring event propagation I observed following for two cases:
Case1 Editable : when table description is editable
    - when we hover anywhere within <label> tags, `edit` icon becomes visible and changes its color to `violet` , means it gets activated and if we click anywhere inside `<label>` tags it will run `toggleEdit function`.
Case2 Read Only: When table description is un-editable
     - when we hover anywhere within <label> tags,`edit` icon becomes visible in `gray` color, and when we hover over `edit` icon only then it changes its color to `violet`. That means `edit` icon in this case gets activated only when we hover over icon itself.
Later is the case which is expected.
In my understanding, difference arise due to rendering of different tags in both cases. We are using `<button>` for editable description and `overlayTrigger` for un-editable. 
But when I use `<div>` tag instead of `<label>`, `edit button` becomes activated only when we click on `edit` icon (not anywhere inside div tags). Hence when we click on any button inside child component like `update` and `cancel`, `edit` button will not get clicked.
Event Propagation takes place when event listeners for both parent and child component are active on same area, which was happening earlier. So I de-coupled it, now `onclick` event inside child component does not activate event listener of parent component.
I tried to stop event propagation by identifying event origin and checking `event.target.name` inside `toggelEdit` function but it was not able to distinguish between origin. 
I also found out that explicitly stopping event propagation by using functions should be avoided as it may affect other functionalities. This may not happen in our case but it is still not recommended.

Reply:
> I also found out that explicitly stopping event propagation by using functions should be avoided as it may affect other functionalities.

Can you link to the source for this. Overall, with web technologies there are many ways to achieve different things, and cases where some things may or may not be recommended -- but at the end of the day engineers have to weigh the pros and cons of certain recommendations within the context of their specific use cases and decide what is best for the application. For example, some questions to ask in this case would be what functionalities would stopping the event propagation affect, and do we care about those functionalities for this component?

As it stands, using a `label` here is also a best practice that we should maintain, so let's consider any other options (i.e. if there is some other fix you can identify without using event propagation), and discuss the pros and cons of each one.

Reply:
Here are sources of that give this recommendation [1](https://css-tricks.com/dangers-stopping-event-propagation/) , [2](https://javascript.info/bubbling-and-capturing). But yes you are right, we have to weigh pros and cons at the end.

Reply:
I saw [this](https://javascript.info/bubbling-and-capturing) article which suggests I can compare `event.target` and `event.currentTarget` to know from where event has bubbled., but this is also not working for in our case, both values are same for different cases.